### PR TITLE
 TST: Add test case for GH14080 for overflow exception

### LINKED ISSUE #14080: Adding big offset to timedelta generates a python crash
##### In:

```
import pandas as pd
from pandas.tseries.frequencies import to_offset

d = pd.Timestamp("2000/1/1")
d + to_offset("D")*100**25
```
##### Out:

**=> python crash**

Fatal Python error: Cannot recover from stack overflow.

Current thread 0x00002b00 (most recent call first):
  File "C:\Users\geoffroy.destaintot\Miniconda3\envs\pd-0.18\lib\site-packages\pandas\tseries\offsets.py", line 2526 in delta
  File "C:\Users\geoffroy.destaintot\Miniconda3\envs\pd-0.18\lib\site-packages\pandas\tseries\offsets.py", line 2535 in apply
  File "C:\Users\geoffroy.destaintot\Miniconda3\envs\pd-0.18\lib\site-packages\pandas\tseries\offsets.py", line 2493 in **add**
  File "C:\Users\geoffroy.destaintot\Miniconda3\envs\pd-0.18\lib\site-packages\pandas\tseries\offsets.py", line 390 in **radd**
  File "C:\Users\geoffroy.destaintot\Miniconda3\envs\pd-0.18\lib\site-packages\pandas\tseries\offsets.py", line 2535 in apply
  File "C:\Users\geoffroy.destaintot\Miniconda3\envs\pd-0.18\lib\site-packages\pandas\tseries\offsets.py", line 2493 in **add**
  File "C:\Users\geoffroy.destaintot\Miniconda3\envs\pd-0.18\lib\site-packages\pandas\tseries\offsets.py", line 390 in **radd**
...
#### Expected Output

Satisfactory behaviour when using python timedeltas:
##### In:

```
import datetime as dt
import pandas as pd
from pandas.tseries.frequencies import to_offset

d = pd.Timestamp("2000/1/1")
d + dt.timedelta(days=1)*100**25
```
##### Out:

**=> python error**

Traceback (most recent call last):
  File "C:/Users/geoffroy.destaintot/Documents/Local/Informatique/Projets/2016-08-django-debug/to_offset_bug.py", line 11, in <module>
    d + dt.timedelta(days=1)_100_*25
OverflowError: Python int too large to convert to C long
#### output of `pd.show_versions()`

(same behaviour with pandas 0.17.1, 0.16.2, 0.15.2)
## INSTALLED VERSIONS

commit: None
python: 3.5.2.final.0
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 69 Stepping 1, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None

pandas: 0.18.1
nose: None
pip: 8.1.2
setuptools: 25.1.6
Cython: None
numpy: 1.11.1
scipy: None
statsmodels: None
xarray: None
IPython: None
sphinx: None
patsy: None
dateutil: 2.5.3
pytz: 2016.6.1
blosc: None
bottleneck: None
tables: None
numexpr: None
matplotlib: None
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
httplib2: None
apiclient: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: None
boto: None
pandas_datareader: None

### PR DESCRIPTION
**Pull request by eoveson**
- [X] closes #14080 
- [X] tests added / passed
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] Unsure if I should modify whatsnew if it is just adding a test?

### CODE REVIEW

File: pandas/tests/scalar/timestamp/test_arithmetic.py [hunk 1]
```diff
@@ -7,11 +7,12 @@
 import pandas.util.testing as tm
 from pandas.compat import long
 from pandas.tseries import offsets
+from pandas.tseries.frequencies import to_offset
 from pandas import Timestamp, Timedelta
 
 
 class TestTimestampArithmetic(object):
-    def test_overflow_offset(self):
+    def test_overflow_offset1(self):
```
Comment (mroeschke) [inline pandas/tests/scalar/timestamp/test_arithmetic.py]: Can you keep the original name of this test.

File: pandas/tests/scalar/timestamp/test_arithmetic.py [hunk 2]
```diff
@@ -27,6 +28,33 @@ def test_overflow_offset(self):
         with pytest.raises(OverflowError):
             stamp - offset
 
+    def test_overflow_offset2(self):
```
Comment (mroeschke) [inline pandas/tests/scalar/timestamp/test_arithmetic.py]: Hmm I see why you did the renaming now. Can you just combine this test and the one above. It looks like there a frew overlapping tests, so if there's anything new being tested just append it to the test above (with the issue number).

Comment (gfyoung) [reply pandas/tests/scalar/timestamp/test_arithmetic.py]: * Let's also check error messages while we're at it here (i.e. `pytest.raises(OverflowError, match=...)`)
*  In the interests of greater modularity in the test base, I would prefer to split the tests into one checking on the expected errors, and one checking the expressions that shouldn't overflow.

Keep in mind that for large tests like these, a single failure prevents us from checking anything afterwards, so the smaller and modular the tests, the better.

Comment (WillAyd) [reply pandas/tests/scalar/timestamp/test_arithmetic.py]: To add to what @gfyoung is saying we typically would create a test for working assertions and another one with `_raises` appended to the test name which tests the failures. You'll see this in other modules so probably a good standard to follow here

Comment (eoveson) [reply pandas/tests/scalar/timestamp/test_arithmetic.py]: Thanks all, sounds good.  I will plan on breaking up into one test with expressions that shouldn't overflow, and one that does (with _raises appended).  Note that the test that covers does-overflow will be a bit long. It seems important to test the specific cases from those issues though. But not sure if I should further break it up (have a look and let me know).

File: pandas/tests/scalar/timestamp/test_arithmetic.py (hunk 1 above)
Comment (eoveson) [reply pandas/tests/scalar/timestamp/test_arithmetic.py]: Split up the tests per the other comment..

**Review by gfyoung — APPROVED** [review]
The tests look good now!  LGTM.

**Comment by jreback** [comment]
thanks @eoveson

**Pull request merged**